google-cloud-bigtable/src/test/java/com/google/cloud/bigtable/admin/v2/it/BigtableCmekIT.java

```diff
@@ -0,0 +1,147 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.bigtable.admin.v2.it;
+
+import static com.google.common.truth.Truth.assertThat;
+import static com.google.common.truth.TruthJUnit.assume;
+
+import com.google.api.gax.rpc.NotFoundException;
+import com.google.cloud.bigtable.admin.v2.BigtableInstanceAdminClient;
+import com.google.cloud.bigtable.admin.v2.BigtableTableAdminClient;
+import com.google.cloud.bigtable.admin.v2.models.Backup;
+import com.google.cloud.bigtable.admin.v2.models.CreateBackupRequest;
+import com.google.cloud.bigtable.admin.v2.models.CreateInstanceRequest;
+import com.google.cloud.bigtable.admin.v2.models.CreateTableRequest;
+import com.google.cloud.bigtable.admin.v2.models.EncryptionInfo;
+import com.google.cloud.bigtable.admin.v2.models.StorageType;
+import com.google.cloud.bigtable.common.Status;
+import com.google.cloud.bigtable.common.Status.Code;
+import com.google.cloud.bigtable.test_helpers.env.AbstractTestEnv;
+import com.google.cloud.bigtable.test_helpers.env.EmulatorEnv;
+import com.google.cloud.bigtable.test_helpers.env.TestEnvRule;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Logger;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.threeten.bp.Instant;
+import org.threeten.bp.temporal.ChronoUnit;
+
+@RunWith(JUnit4.class)
+public class BigtableCmekIT {
+  private static final String CMEK_KMS_KEY_PROPERTY_NAME = "bigtable.kms_key_name";
+
+  @ClassRule public static TestEnvRule testEnvRule = new TestEnvRule();
+
+  private static String kmsKeyName;
+  private String instanceId;
+  private String clusterId;
+
+  private BigtableInstanceAdminClient instanceAdmin;
+  private BigtableTableAdminClient tableAdmin;
+
+  @BeforeClass
+  public static void validatePlatform() {
+    assume()
+        .withMessage("Emulator doesn't support on CMEK")
+        .that(testEnvRule.env())
+        .isNotInstanceOf(EmulatorEnv.class);
+
+    kmsKeyName = System.getProperty(CMEK_KMS_KEY_PROPERTY_NAME);
+
+    assume().withMessage("%s property is not set, skipping CMEK test", kmsKeyName);
+  }
+
+  @Before
+  public void setUp() throws IOException {
+    instanceId = AbstractTestEnv.TEST_INSTANCE_PREFIX + Instant.now().getEpochSecond();
+    clusterId = instanceId + "-c1";
```

Comment:
Shoudnt this be `TEST_CLUSTER_PREFIX`?

Reply:
This works too I guess. Instance ID suffixed with "-c1" and "-c2" for the 2 cluster names?